sci-libs/cantera/cantera-2.4.0.ebuild

```diff
@@ -0,0 +1,144 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_{4,5,6,7}} )
+
+inherit desktop python-r1 scons-utils toolchain-funcs
+
+DESCRIPTION="Object-oriented tool suite for chemical kinetics, thermodynamics, and transport"
+HOMEPAGE="http://www.cantera.org"
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+SRC_URI="https://github.com/Cantera/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+IUSE="+cpp +cti fortran pch python static-libs test"
+
+## Python2 is required by '<scons-3.0' to work.
+## Python3 automatic detection is used by cantera before compilling
+## if python3_package isn't set no "n".
+
+REQUIRED_USE="
+	|| ( cpp python )
+	cti? ( ${PYTHON_REQUIRED_USE} )
+	fortran? ( cpp )
+	python? ( cti )
+	static-libs? ( cpp )
+	?? ( python_targets_python3_4 python_targets_python3_5 python_targets_python3_6 python_targets_python3_7 )
```

Comment:
Don't do `??` for Python stuff, don't hardcode any `python_targets` flags, **ever**). Always `^^` and `python_gen_useflags` function; though we'll probably discover you shouldn't be doing this anyway.

Reply:
We have at least two major packages like this, and we had py2+py3 support in them like this. However, the maintainers decided to reduce it to py2||py3 shortly afterwards because the resulting hassle for users didn't justify the real need for two simultaneous interpreters. So please just switch to `python-single-r1` and forget about py2+py3 because it's not worth it.

Reply:
Ok, then I remove python2 dependency and leave only python3.

What about SCons? When dev-util/scons-3.0.1-r100 is planned to become  *-r4? Or all of scons-3.0.1 ebuilds with python3 support will have *-r1xx revisions?

If last is true then I add dependency of >=dev-util/scons-3.0.1-r100

Reply:
Oh, i'm going to test SCons with 3.7 today.

Reply:
I will waiting for it in portage tree and then will update commit after some additional build tests.

Reply:
I'm afraid SCons doesn't work reasonably with py3.7, so please just remove 3.7 support. You may also want to point out to upstream that SCons is practically dead these days and switching to a maintained build system would be a good option.

Reply:
I see [some commits](https://github.com/SCons/scons/search?o=desc&p=1&q=python+3.7&s=committer-date&type=Commits) related with python 3.7 in SCons upstream among other recent commits. They even recently added the python 3.7 to their Travis CI. Is there a build problem of SCons wtith python 3.7?

I don't expect that upstream of Cantera will make next major update until the 2020 (when support of python 2.7 will be dropped finally).

And as the version of python used to build SCons doesn't affect the version of python to build Cantera itself then could I currently left the `soft` dependency like `dev-util/scons` instead of `hard` like  `dev-util/scons[${PYTHON_USEDEP}]`? It because user currently still can uses SCons (with python2.7 or arbitrary python3_x) for other tools within system and because Scons support multiple python_targets itself instead of single_python_targets. 

Or do you insist of hard `dev-util/scons[${PYTHON_USEDEP}]` dependency? There are still some package in portage tree that depends on `inherit scons-utils` and that only have PYTHON_COMPAT=( python2_7 ).

Reply:
Commits mean little when they're not accompanied by a release, and SCons hasn't seen a release for almost a year. The previous releases were also pretty late to the party. I don't really have time to try to find relevant commits in their git log.

Yes, I insist on PYTHON_USEDEP. Since SCons gained py3 support, this is how we need to handle it. Technically, you could try to use the any-r1 API combined with main API if upstream somehow makes SCons-calling distinct enough to allow it to use another Python version. However, this will be horribly hard if not entirely impossible, and I can't afford the time needed to figure it out.

Reply:
Sorry for delay.

Done.
+ Switch to IUSE="... cxx +python ..." instead of "... +cxx python ..."  used early as default flags.